### PR TITLE
fix(ui): filtering repos by active status

### PR DESCRIPTION
**Pull request by liangfung**
https://jam.dev/c/0ba6d9e6-7472-49df-b79a-b7ade4fd396d

### CODE REVIEW

**Review by wsxiaoys — APPROVED** [review]

File: ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/gitlab-provider-detail.tsx [hunk 1]
```diff
@@ -123,118 +129,239 @@ function toStatusBadge(status: RepositoryProviderStatus) {
...
-}> = ({ data, onDelete, providerStatus }) => {
+}> = ({ onDelete, providerStatus, providerId }) => {
+  const [page, setPage] = React.useState(1)
+  const {
+    repositories: inactiveRepositories,
+    setRepositories: setInactiveRepositories,
+    isAllLoaded: isInactiveRepositoriesLoaded
+  } = useAllInactiveRepositories(providerId)
+
+  const doFetchRepositories = (
```
Comment (wwayne) [inline ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/gitlab-provider-detail.tsx:144]: Maybe the name can just be `fetchRepositories`?

File: ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/gitlab-provider-detail.tsx [hunk 2]
```diff
@@ -123,118 +129,239 @@ function toStatusBadge(status: RepositoryProviderStatus) {
...
+    }).then(res => {
+      if (res?.data?.updateGitlabProvidedRepositoryActive) {
+        setInactiveRepositories([...inactiveRepositories, repo])
+        const nextPage = isLastItem ? page - 1 : page
+        doLoadPage(nextPage || 1)
+      }
     })
   }
 
+  const doLoadPage = async (pageNo: number) => {
```
Comment (wwayne) [inline ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/gitlab-provider-detail.tsx:201]: `loadPage` ?

**Review by wwayne — APPROVED** [review]

File: ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/gitlab-provider-detail.tsx (hunk 1 above)
Comment (liangfung) [reply ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/gitlab-provider-detail.tsx]: renamed

File: ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/gitlab-provider-detail.tsx (hunk 2 above)
Comment (liangfung) [reply ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/gitlab-provider-detail.tsx]: renamed

**Pull request merged**